### PR TITLE
Change Objective C class references to using macros.

### PR DESCRIPTION
**Pull request by dmaclach**
Allows easy replacing of class names in generated code using #define to redefine class names at compile time.

### CODE REVIEW

File: objectivec/GPBUtilities_PackagePrivate.h [hunk 1]
```diff
@@ -36,13 +36,22 @@
...
-// string. See README.google for more information.
+// string.
 #define GPBStringify(S) #S
 #define GPBStringifySymbol(S) GPBStringify(S)
 
 #define GPBNSStringify(S) @#S
 #define GPBNSStringifySymbol(S) GPBNSStringify(S)
 
+// Macros for generating a Class from a class name. These are used in
+// the generated PB descriptor classes wherever an Objective C class
```
Comment (thomasvl) [inline objectivec/GPBUtilities_PackagePrivate.h]: PB -> GPB ?

File: src/google/protobuf/compiler/objectivec/objectivec_file.cc [hunk 2]
```diff
@@ -432,12 +432,9 @@ void FileGenerator::GenerateSource(io::Printer *printer) {
       "\n");
   if (!fwd_decls.empty()) {
     printer->Print(
-        "#pragma mark - Objective C Class references\n"
-        "// This somewhat arcane code forces linkage of classes from static archives by\n"
-        "// adding a concrete reference to the classes.\n"
-        "// We don't use `[Foo class]` because we need a static value for our initializer.\n"
```
Comment (thomasvl) [inline src/google/protobuf/compiler/objectivec/objectivec_file.cc]: These last three lines might still be useful to explain to someone, up to you if you want to keep them.

File: objectivec/GPBUtilities_PackagePrivate.h (hunk 1 above)
Comment (dmaclach) [reply objectivec/GPBUtilities_PackagePrivate.h]: Done (and in the comment above that I probably copied from)

File: src/google/protobuf/compiler/objectivec/objectivec_file.cc (hunk 2 above)
Comment (dmaclach) [reply src/google/protobuf/compiler/objectivec/objectivec_file.cc]: Added the important clarifying line IMHO.

**Pull request merged**